### PR TITLE
Updating Meshery Adapter for Kuma to Go version to 1.23 

### LINKED ISSUE #441: [Chore] Upgrade to Golang 1.23.4
#### Current Behavior
Project is running golang 1.21.8.

#### Desired Behavior
Project need updated to golang 1.23.4

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 📝 [Meshery Adapters Spreadsheet](https://docs.google.com/spreadsheets/d/1rGGpSXC68iDJzNRU-qZIRQsZUwkt1qLFg7JCs4pfvbU/edit?usp=sharing)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by devhindo**
Just updated Dockerfile and Workflows to go v23 and everything seems to work fine

fixes #441

### CODE REVIEW

**Review by leecalcote — COMMENTED** [review]
It'd be good to upgrade to the v5 version of the go action.

So many uses of that go action seems redundant and like unnecessary overhead and an opportunity for consolidation.

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -19,9 +19,9 @@ jobs:
       with:
```
Comment (devhindo) [inline .github/workflows/build-and-release.yml:19]: should we also update actions/

**Comment by devhindo** [comment]
Regarding line 18 of .github/workflows/build-and-release.yml:
should we also update [actions/checkout@v4](https://github.com/actions/checkout) ?

**Comment by devhindo** [comment]
@leecalcote fixed some golangci-lint suggestions and updated the go action version. waiting for you opinion!

**Comment by leecalcote** [comment]
Running build workflows....

**Comment by devhindo** [comment]
updated: dominikh/staticcheck-action@v1 due to deprecated feature [(source)](https://github.com/dominikh/go-tools/issues/1387#issuecomment-1486076852)

**Comment by devhindo** [comment]
please rerun @leecalcote

**Comment by leecalcote** [comment]
Re-running

**Comment by leecalcote** [comment]
I wonder if this action, https://github.com/dominikh/staticcheck-action,  can be replaced by another action.

**Comment by devhindo** [comment]
we were just using a deprecated version. please rerun @leecalcote 
I'm also wondering what's the issue with [this](https://github.com/meshery/meshery-kuma/actions/runs/12303031223/job/34338304177) action as the logs doesn't show anything userful.

**Comment by leecalcote** [comment]
Let's push forth with a merge here and see where we stand afterwards....

**Pull request merged**